src/Common/Method/DeprecatedObjectUsageJsonDecodeRule.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Method;
+
+use ArchitectureSniffer\Common\DeprecationTrait;
+use PHPMD\AbstractNode;
+use PHPMD\AbstractRule;
+use PHPMD\Node\ASTNode;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\ClassAware;
+
+class DeprecatedObjectUsageJsonDecodeRule extends AbstractRule implements ClassAware
+{
+    use DeprecationTrait;
+
+    /**
+     * @var string
+     */
+    protected const METHOD_NAME = 'decodeJson';
+
+    /**
+     * @var string
+     */
+    protected const CORRECT_VALUE = 'true';
+
+    /**
+     * @var string
+     */
+    protected const RULE_DEPRECATED_USAGE_FOUND = 'Param #2 `$assoc` of `decodeJson` must be `true` as return of type `object` is not accepted. Method: %s.';
```

Comment:
not accepted => deprecated
maybe
But yeah, no big deal, sounds OK to me.

Reply:
I'd say, `return value of type 'object'` sounds better than `return of type 'object'`.

Reply:
Fixed both :)